bindings/wasm/src/account/wasm_account.rs

```diff
@@ -0,0 +1,360 @@
+  pub fn delete_identity(self) -> Promise {
+    //ToDo: test once `load_identity(..)` is implemented.
+
+    // Get IotaDID and storage from the account.
+    let account: Rc<WasmRefCell<Account>> = self.0;
+    let did: IotaDID = account.as_ref().borrow().did().to_owned();
+    let storage: Arc<dyn Storage> = account.as_ref().borrow().storage_arc();
+
+    // Drop account should release the DIDLease because we cannot take ownership of the Rc.
+    // Note that this will still fail if anyone else has a reference to the Account.
+    std::mem::drop(account);
+
+    future_to_promise(async move {
+      // Create a new account since `delete_identity` consumes it.
+      let account: Result<Account> = AccountBuilder::new()
+        .storage(AccountStorage::Custom(storage))
+        .load_identity(did)
+        .await
+        .wasm_result();
+
+      match account {
+        Ok(a) => a.delete_identity().await.wasm_result().map(|_| JsValue::undefined()),
+        Err(e) => Err(e),
+      }
+    })
+  }
+
+  /// Push all unpublished changes to the tangle in a single message.
+  #[wasm_bindgen]
+  pub fn publish(&mut self, publish_options: Option<WasmPublishOptions>) -> Promise {
+    let account = self.0.clone();
+    if let Some(publish_options) = publish_options {
+      let mut options: PublishOptions = PublishOptions::new();
+
+      if let Some(force_integration) = publish_options.forceIntegrationUpdate() {
+        options = options.force_integration_update(force_integration);
+      }
+
+      if let Some(sign_with) = publish_options.signWith() {
+        let s: String = sign_with;
+        options = options.sign_with(s);
+      };
+      future_to_promise(async move {
+        account
+          .as_ref()
+          .borrow_mut()
+          .publish_with_options(options)
+          .await
+          .map(|_| JsValue::undefined())
+          .wasm_result()
+      })
+    } else {
+      future_to_promise(async move {
+        account
+          .as_ref()
+          .borrow_mut()
+          .publish()
+          .await
+          .map(|_| JsValue::undefined())
+          .wasm_result()
+      })
+    }
+  }
+
+  /// Signs a {@link Credential} with the key specified by `fragment`.
+  #[wasm_bindgen(js_name = createSignedCredential)]
+  pub fn create_signed_credential(
+    &self,
+    fragment: String,
+    credential: &WasmCredential,
+    signature_options: &WasmSignatureOptions,
+  ) -> PromiseCredential {
+    let account = self.0.clone();
+    let mut cred: Credential = credential.0.clone();
+    let options: SignatureOptions = SignatureOptions::from(signature_options);
+
+    let promise: Promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .sign(fragment.as_str(), &mut cred, options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()?;
+      JsValue::from_serde(&cred).wasm_result()
+    });
+    promise.unchecked_into::<PromiseCredential>()
```

Comment:
You could reduce duplication in these `create_signed_*` methods, as an example:

```rust
  /// Signs a {@link Document} with the key specified by `fragment`.
  #[wasm_bindgen(js_name = createSignedDocument)]
  pub fn create_signed_document(
    &self,
    fragment: String,
    document: &WasmDocument,
    signature_options: &WasmSignatureOptions,
  ) -> PromiseDocument {
    self
      .create_signed(fragment, document.0.clone(), signature_options)
      .unchecked_into::<PromiseDocument>()
  }

  fn create_signed<U>(&self, fragment: String, mut data: U, signature_options: &WasmSignatureOptions) -> Promise
  where
    U: serde::Serialize + SetSignature + 'static,
  {
    let account = self.0.clone();
    let options: SignatureOptions = SignatureOptions::from(signature_options);

    future_to_promise(async move {
      account
        .as_ref()
        .borrow_mut()
        .sign(fragment.as_str(), &mut data, options)
        .await
        .map(|_| JsValue::undefined())
        .wasm_result()?;
      JsValue::from_serde(&data).wasm_result()
    })
  }
```

Unless I'm missing something, you should be able to call `create_signed` from the other methods as well.